presto-tpcds/src/main/java/com/facebook/presto/tpcds/TpcdsNodePartitioningProvider.java

```diff
@@ -49,21 +47,13 @@ public TpcdsNodePartitioningProvider(NodeManager nodeManager, int splitsPerNode)
     }
 
     @Override
-    public Map<Integer, Node> getBucketToNode(ConnectorTransactionHandle transactionHandle, ConnectorSession session, ConnectorPartitioningHandle partitioningHandle)
+    public ConnectorBucketNodeMap getBucketNodeMap(ConnectorTransactionHandle transactionHandle, ConnectorSession session, ConnectorPartitioningHandle partitioningHandle)
     {
         Set<Node> nodes = nodeManager.getRequiredWorkerNodes();
         checkState(!nodes.isEmpty(), "No TPCDS nodes available");
 
         // Split the data using split and skew by the number of nodes available.
-        ImmutableMap.Builder<Integer, Node> bucketToNode = ImmutableMap.builder();
-        int partNumber = 0;
-        for (Node node : nodes) {
-            for (int i = 0; i < splitsPerNode; i++) {
-                bucketToNode.put(partNumber, node);
-                partNumber++;
-            }
-        }
-        return bucketToNode.build();
+        return new ConnectorBucketNodeMap(nodes.size() * splitsPerNode);
```

Comment:
just to make sure, can `nodes.size() * splitsPerNode` overflow (they are both integers)?